api/controllers/case.js

```diff
@@ -191,7 +195,24 @@ router.put('/:caseId', function editCaseById (req, res) {
     }).then(function(data){
         let the_case = data[0];
         the_case.authors = data[1]; // authors
-        the_case.location = data[2]; // geolocation
+        let attachments = data[2]; // files and images
+        the_case.other_images = []
+        the_case.files = []
+        attachments.forEach(function(att){
+            if (att.type == 'file'){
+                the_case.files.push(att);
+            }else if (att.type == 'image'){
+                if (att.is_lead){
+                    the_case.lead_image = att;
+                }else{
+                    the_case.other_images.push(att);
+                }
+            }
+        });
+        the_case.methods = data[3]
```

Comment:
This is another place where I don't like the fact that we end up having to match the order of SQL statements in the `batch` block with the order of unpacking the returned data in the `then()` block.  Don't know if there's better syntax for that, but it feels brittle in the face of changing code.